### PR TITLE
Release/v0.4.3

### PR DESCRIPTION
**Pull request by Sha3nS**
# Goals of PR

Core changes:
merge release/v04.3 into main

Notes:
no

Related Issues:
no

### CODE REVIEW

**Review by guoshijiang — DISMISSED** [review]

File: l2geth/core/vm/evm.go [hunk 1]
```diff
@@ -425,6 +425,9 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 		return nil, common.Address{}, gas, ErrInsufficientBalance
 	}
 	nonce := evm.StateDB.GetNonce(caller.Address())
+	if nonce+1 < nonce {
+		return nil, common.Address{}, gas, ErrNonceUintOverflow
+	}
 	evm.StateDB.SetNonce(caller.Address(), nonce+1)
```
Comment (Sha3nS) [inline l2geth/core/vm/evm.go:431]: what‘s the purpose of this code?

File: mt-batcher/cmd/mt-batcher/main.go [hunk 2]
```diff
@@ -2,12 +2,15 @@ package main
 
 import (
 	"fmt"
+	"os"
+
+	"github.com/urfave/cli"
+
```
Comment (Sha3nS) [inline mt-batcher/cmd/mt-batcher/main.go:8]: should delete this line

File: mt-challenger/cmd/mt-challenger/main.go [hunk 3]
```diff
@@ -2,12 +2,15 @@ package main
 
 import (
 	"fmt"
+	"os"
+
+	"github.com/urfave/cli"
+
```
Comment (Sha3nS) [inline mt-challenger/cmd/mt-challenger/main.go:8]: should delete this line

File: mt-challenger/mt_challenger.go [hunk 4]
```diff
@@ -4,15 +4,17 @@ import (
 	"context"
 	"time"
 
+	"github.com/urfave/cli"
+
```
Comment (Sha3nS) [inline mt-challenger/mt_challenger.go:8]: should delete this line

File: packages/contracts/contracts/L2/predeploys/BVM_MANTLE.sol [hunk 5]
```diff
@@ -6,6 +6,9 @@ import { Lib_PredeployAddresses } from "../../libraries/constants/Lib_PredeployA
 
 /* Contract Imports */
 import { L2StandardERC20 } from "../../standards/L2StandardERC20.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
```
Comment (Sha3nS) [inline packages/contracts/contracts/L2/predeploys/BVM_MANTLE.sol:9]: delete import

File: packages/contracts/deploy-config/goerli-testnet.ts [hunk 6]
```diff
@@ -3,8 +3,6 @@ const config = {
   l1BlockTimeSeconds: 15,
   l2BlockGasLimit: 15_000_000,
   l2ChainId: 5001,
-  updateGaslimitBlock: 222073,
-  eigenDaBlock: 8280000,
   ctcL2GasDiscountDivisor: 32,
```
Comment (Sha3nS) [inline packages/contracts/deploy-config/goerli-testnet.ts:6]: what's the related config now?

File: packages/sdk/src/utils/contracts.ts [hunk 7]
```diff
@@ -48,6 +48,20 @@ const NAME_REMAPPING = {
...
+ */
+export const L1_MNT_ADDRESS: {
+  [ChainID in L2ChainID]: string
+} = {
+  [L2ChainID.MANTLE]: "0x3c3a81e81dc49A522A592e7622A7E711c06bf354",
+  [L2ChainID.MANTLE_KOVAN]: "0x3c3a81e81dc49A522A592e7622A7E711c06bf354",
+  [L2ChainID.MANTLE_GOERLIQA]: "0x3c3a81e81dc49A522A592e7622A7E711c06bf354",
+  [L2ChainID.MANTLE_TESTNET]: "0xc1dC2d65A2243c22344E725677A3E3BEBD26E604",
+  [L2ChainID.MANTLE_HARDHAT_LOCAL]: "0x3c3a81e81dc49A522A592e7622A7E711c06bf354",
+  [L2ChainID.MANTLE_HARDHAT_DEVNET]: "0x3c3a81e81dc49A522A592e7622A7E711c06bf354"
```
Comment (Sha3nS) [inline packages/sdk/src/utils/contracts.ts:62]: is these correct?

File: tss/common/utils.go [hunk 8]
```diff
@@ -8,6 +8,7 @@ import (
 	"time"
 
 	"github.com/btcsuite/btcd/btcec"
+
```
Comment (Sha3nS) [inline tss/common/utils.go:11]: should delete this line

File: packages/contracts/deploy-config/goerli-testnet.ts (hunk 6 above)
Comment (Tri-stone) [reply packages/contracts/deploy-config/goerli-testnet.ts:6]: These two config has been set as below:
- https://github.com/mantlenetworkio/mantle/blob/0b7733c10a44e93edbf49b27e056ebe2e5279c63/l2geth/core/upgrade/config.go#L54
- https://github.com/mantlenetworkio/mantle/blob/0b7733c10a44e93edbf49b27e056ebe2e5279c63/l2geth/core/upgrade/config.go#L55

File: packages/sdk/src/utils/contracts.ts (hunk 7 above)
Comment (Tri-stone) [reply packages/sdk/src/utils/contracts.ts:62]: > is these correct?

correct

File: packages/contracts/deploy-config/goerli-testnet.ts (hunk 6 above)
Comment (Tri-stone) [reply packages/contracts/deploy-config/goerli-testnet.ts:6]: > what's the related config now?

These two config has been set as below:
- https://github.com/mantlenetworkio/mantle/blob/0b7733c10a44e93edbf49b27e056ebe2e5279c63/l2geth/core/upgrade/config.go#L54
- https://github.com/mantlenetworkio/mantle/blob/0b7733c10a44e93edbf49b27e056ebe2e5279c63/l2geth/core/upgrade/config.go#L55

File: mt-batcher/cmd/mt-batcher/main.go (hunk 2 above)
Comment (guoshijiang) [reply mt-batcher/cmd/mt-batcher/main.go:8]: This conforms to the specification of our go format imports

File: mt-challenger/cmd/mt-challenger/main.go (hunk 3 above)
Comment (guoshijiang) [reply mt-challenger/cmd/mt-challenger/main.go:8]: This conforms to the specification of our go format imports

File: mt-challenger/mt_challenger.go (hunk 4 above)
Comment (guoshijiang) [reply mt-challenger/mt_challenger.go:8]: This conforms to the specification of our go format imports

File: tss/common/utils.go (hunk 8 above)
Comment (guoshijiang) [reply tss/common/utils.go:11]: This conforms to the specification of our go format imports

File: l2geth/core/vm/evm.go (hunk 1 above)
Comment (Tri-stone) [reply l2geth/core/vm/evm.go:431]: > what‘s the purpose of this code?

https://github.com/mantlenetworkio/mantle/pull/1292

Comment (Sha3nS) [reply l2geth/core/vm/evm.go:431]: Uint64 overflow

File: packages/contracts/contracts/L2/predeploys/BVM_MANTLE.sol (hunk 5 above)
Comment (Sha3nS) [reply packages/contracts/contracts/L2/predeploys/BVM_MANTLE.sol:9]: will solved in this PR: https://github.com/mantlenetworkio/mantle/pull/1301

**Review by guoshijiang — APPROVED** [review]

**Review by Tri-stone — APPROVED** [review]

**Review by wukongcheng — APPROVED** [review]

**Pull request merged**